src/main/java/pl/javastart/task/PalindromeService.java

```diff
@@ -4,6 +4,36 @@ public class PalindromeService {
 
     // nie zmieniaj sygnatury tej metody. Jest ona testowana w PalindromeServiceTest
     public boolean isPalindrome(int[] array) {
+        if (array == null) {
+            return false;
+        } else {
+            int[] reverseArray = new int[array.length];
+            for (int i = 0; i < reverseArray.length; i++) {
+                reverseArray[i] = array[array.length - i - 1];
+            }
+            if (array.length == 0 || isFilledByZeros(array)) {
+                return false;
+            } else if (areEquals(array, reverseArray)) { // możliwe prostsze rozwiązanie (Arrays.equals(array, reverseArray))
+                return true;
+            }
```

Comment:
Rozwiązanie z tworzeniem dodatkowej tablicy z elementami w odwrotnej kolejności jest ok, ale jest mało wydajne.
Zauważ, że jeżeli mamy np. tablicę, w której jest 100tys elementów: `[5, 2, ..., 3, 5]`, to już na drugim elemencie jesteśmy w stanie stwierdzić, że tablica nie jest palindromem.
Spróbuj zapisać pętlę, w której porównasz 1 i ostatni element, 2 i przedostatni element itd. Gdy elementy nie będą się zgadzały, to wiesz, że to nie palindrom.